### PR TITLE
Fix SHARPOP precedence

### LINKED ISSUE #1507: `location##streets[1]` precedence is wrong
`(location##streets)[1]` is the right precedence. But it formats to `location##streets[1]`. The first form is the desirable precedence, but the latter form is the desirable visual.

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes #1507 #774

Not really sure how to write tests for this one.

I verified the following expressions manually:

```shell
$ echo 'foo##bar.baz' | ./_build/install/default/bin/refmt -p ml
;;(foo ## bar).baz

$ echo 'foo##bar[1]' | ./_build/install/default/bin/refmt -p ml
;;(foo ## bar).(1)

$ echo 'foo##bar[0]##baz[1]' | ./_build/install/default/bin/refmt -p ml
;;(((foo ## bar).(0)) ## baz).(1)
```

### CODE REVIEW

**Comment by anmonteiro** [comment]
Just fixed the issue pointed out in my last comment with the latest commit.

File: src/reason-parser/reason_parser.mly [hunk 1]
```diff
@@ -1209,6 +1209,7 @@ let package_type_of_module_type pmty =
 %token SEMISEMI
 %token SHARP
 %token <string> SHARPOP
+%token SHARPOP_WITH_EQUAL
```
Comment (IwanKaramazow) [inline src/reason-parser/reason_parser.mly]: can you rename this to `SHARPEQUAL`? `SHARPOP_WITH_EQUAL` conveys `# with = `, if you look at the other token names.

Comment (anmonteiro) [reply src/reason-parser/reason_parser.mly]: definitely

File: src/reason-parser/reason_lexer.mll [hunk 2]
```diff
@@ -481,10 +481,15 @@ rule token = parse
   | "#=<" {
     (* Allow parsing of foo#=<bar /> *)
     set_lexeme_length lexbuf 2;
-    SHARPOP("#=")
+    SHARPOP_WITH_EQUAL
   }
   | "#" operator_chars+
-      { SHARPOP(lexeme_operator lexbuf) }
+      { let l = lexeme_operator lexbuf in
+        if l = "#=" then
```
Comment (IwanKaramazow) [inline src/reason-parser/reason_lexer.mll]: I wonder if we should add `| "#=" { SHARPEQUAL }` for clarity in the lexer. No hard opinions here.

**Comment by chenglou** [comment]
M-m-merged!

**Pull request merged**